### PR TITLE
Add an Antora-based site generator

### PR DESCRIPTION
**Pull request by Mogztter**

### CODE REVIEW

**Comment by mraible** [comment]
@Mogztter I noticed an issue today on the [generated site](https://infoq-mini-book.netlify.app/). The main landing page has a title (that shows in the browser tab) of "Untitled :: InfoQ Mini Book". Is it possible to fix that?

File: src/docs/asciidoc/chapters/chapter1.adoc [hunk 1]
```diff
@@ -1,2 +1,3 @@
 [[chapter-1]]
 = Chapter 1
+:page-pagination: next
```
Comment (mraible) [inline src/docs/asciidoc/chapters/chapter1.adoc:3]: What do the different values for `page-pagination` mean?

File: src/docs/asciidoc/nav.adoc [hunk 2]
```diff
@@ -0,0 +1,9 @@
+* xref:dedication.adoc[]
```
Comment (mraible) [inline src/docs/asciidoc/nav.adoc:1]: Is [`colophon.adoc`](https://github.com/mraible/infoq-mini-book/blob/main/src/docs/asciidoc/chapters/colophon.adoc) missing from this file on purpose?

File: src/docs/asciidoc/nav.adoc [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+* xref:acknowledgements.adoc[]
```
Comment (mraible) [inline src/docs/asciidoc/nav.adoc:2]: I recently learned this is spelled "acknowledgments". I can rename the file after this is merged.

**Comment by mraible** [comment]
@Mogztter I realized today that https://infoq-mini-book.netlify.app does not have a social image. Is it possible to add one? I usually send myself a message in Slack so see if a site has a social image. This URL pulls in a social image and description. 

https://developer.okta.com/blog/2022/02/11/angular-auth0-quickly

9742D963-BD92-4375-8586-3E98F8000EFF

**Comment by Mogztter** [comment]
> I noticed an issue today on the [generated site](https://infoq-mini-book.netlify.app/). The main landing page has a title (that shows in the browser tab) of "Untitled :: InfoQ Mini Book". Is it possible to fix that?

Sure, what should we use instead? Maybe the value of the `book-title` attribute?

> I realized today that [infoq-mini-book.netlify.app](https://infoq-mini-book.netlify.app/) does not have a social image. Is it possible to add one?

Yes, we can introduce a new AsciiDoc attribute named `og-image` and use the value of this attribute in the UI.
You can also define a description and keywords: https://github.com/Mogztter/mini-book-default-ui/blob/f4113c5b7f22c7f217c5b6140af3ea32f2c376a1/src/partials/head-info.hbs#L12-L17

> I usually send myself a message in Slack so see if a site has a social image. This URL pulls in a social image and description.

You can also use https://www.opengraph.xyz/ or https://socialsharepreview.com/

File: src/docs/asciidoc/nav.adoc (hunk 2 above)
Comment (Mogztter) [reply src/docs/asciidoc/nav.adoc:1]: No reason, I can add it

Comment (Mogztter) [reply src/docs/asciidoc/nav.adoc:1]: Done!

File: src/docs/asciidoc/chapters/index.adoc [hunk 4]
```diff
@@ -0,0 +1,3 @@
+:page-title: My Book
```
Comment (Mogztter) [inline src/docs/asciidoc/chapters/index.adoc:1]: ☝🏻 Define a title on the page.
This title will be used as the document title on the generated HTML page: `My Book :: InfoQ Mini Book`

Comment (mraible) [reply src/docs/asciidoc/chapters/index.adoc:1]: Excellent! I like this, but for the Angular Mini-Book, I think it'd render `The Angular Mini Book :: The Angular Mini Book`. I could see using `Home` as the title, but ideally, there'd be a way to just render the title of the book on the main landing page.

Comment (Mogztter) [reply src/docs/asciidoc/chapters/index.adoc]: I think the following should work;

```
= The Angular Mini Book
:page-title: Home
```

The document title on the generated HTML page will be `Home :: The Angular Mini Book` while the title (`<h1>`) on the HTML page will be `The Angular Mini Book`.

**Comment by Mogztter** [comment]
@mraible AsciiDoc files moved to `src/docs/asciidoc/modules/ROOT`. The main benefits are:

- IntelliJ will render images (because the project uses a standard Antora structure)
- We can use the default site generator instead of maintaining a fork (namely `mini-book-site-generator`)

File: src/docs/asciidoc/antora.yml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+name: ROOT
+title: My Book
+version: ~
```
Comment (Mogztter) [inline src/docs/asciidoc/antora.yml:3]: since we are using the default generator with Antora 3, we can use `~` instead of `master`

**Comment by mraible** [comment]
I realized that I can change the URL for chapters by changing the filename. In the past, the filename was never an issue since we never published an HTML version. I like naming the files `chapter1.adoc`, `chapter2.adoc`, etc. to keep them organized. However, I can't help but think the site generator should use the title of the chapter instead. Or maybe there's a way to set the URL at the top of the chapter?

**Comment by Mogztter** [comment]
> I realized that I can change the URL for chapters by changing the filename.

That's correct.

>  In the past, the filename was never an issue since we never published an HTML version. 
I like naming the files chapter1.adoc, chapter2.adoc, etc. to keep them organized. However, I can't help but think the site generator should use the title of the chapter instead. Or maybe there's a way to set the URL at the top of the chapter?

Technically, it should be possible to create a custom Antora extension to use a slug instead of the filename in the URL?
It would be define as an AsciiDoc attribute:

**chapter1.adoc**
```
= Build an Angular App
:slug: build-an-angular-app
```

We could also automatically generate a slug from the title but if we use an AsciiDoc attribute then we can enable this feature only on certain files.

**Comment by Mogztter** [comment]
> I like the idea of a custom Antora extension! :slug: works, but something like :page-url: seems more intuitive.

`:page-url` is a page attribute and can be accessed from the UI template using the UI model (which is not what we want)

https://docs.antora.org/antora/latest/page/page-attributes/

Naming wise, `url` is unclear because it could be an absolute or relative URL whereas, in this case, it's just the "name of the page" segment:

module-page-url

If we want to use Antora terminology, we could use `:pub-page-name:` (pub stands for public or publish) or `:pub-slug:`.
I like `slug` because it makes it clear that the value will be use in the URL and as a result it should be normalized.

Anyway, is it a major issue that we should fix before merging?

**Comment by mraible** [comment]
> Anyway, is it a major issue that we should fix before merging?

No, not at all.

**Pull request merged**